plugins/module_utils/system_access_users_utils.py

```diff
@@ -0,0 +1,855 @@
+    ipsecpsk: Optional[str] = None
+    otp_seed: Optional[str] = None
+    shell: Optional[UserLoginShell] = None
+    uid: Optional[str] = None
+    disabled: bool = False
+    full_name: Optional[str] = None
+    email: Optional[str] = None
+    comment: Optional[str] = None
+    landing_page: Optional[str] = None
+    expires: Optional[str] = None
+    authorizedkeys: Optional[str] = None
+    cert: Optional[str] = None  # TODO is in another xml path
+    apikeys: Optional[List[str]] = None
+    groupname: Optional[List[str]] = None
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, User):
+            return False
+
+        for field in fields(self):
+            if (
+                field.name != "password"
+                and field.name != "uid"
+                and field.name != "otp_seed"
+                and field.name != "apikeys"
+            ):
+                if getattr(self, field.name) != getattr(other, field.name):
+                    return False
+
+        return True
+
+    def __post_init__(self):
+        # Manually define the fields and their expected types
+        enum_fields = {
+            "shell": UserLoginShell,
+        }
+
+        for field_name, field_type in enum_fields.items():
+            value = getattr(self, field_name)
+
+            # Check if the value is a string and the field_type is a subclass of ListEnum
+            if isinstance(value, str) and issubclass(field_type, ListEnum):
+                # Convert string to ListEnum
+                setattr(self, field_name, field_type.from_string(value))
+
+    def set_otp_seed(self, otp_seed: str = None) -> str:
+        """
+        Generates and returns a base32-encoded OTP seed.
+
+        Args:
+            otp_seed (str, optional): Existing OTP seed to encode (default: None).
+
+        Returns:
+            str: Base32-encoded OTP seed.
+
+        If no OTP seed is provided, a random seed is generated and encoded as base32.
+        """
+
+        if otp_seed is None:
+            otp_seed = os.urandom(20)
+
+        return base64.b32encode(otp_seed.encode("utf-8")).decode("utf-8")
+
+    def set_apikeys(self, apikeys: list = None) -> list:
+        """
+        Generates a list of dictionaries, each containing a 'key' and a 'secret'.
+        If apikeys is provided, each element in apikeys is used as the 'key',
+        and a new 'secret' is generated.
+        If apikeys is not provided or is an empty list, a single 'key'-'secret' pair is generated.
+
+        Args:
+            apikeys (list, optional): A list of strings to be used as the 'key' part of each pair.
+
+        Returns:
+            list: A list of dictionaries, where each dictionary has a 'key' and a 'secret'.
+        """
+
+        api_keys = []
+
+        # Check if apikeys is None or contains an empty string
+        if apikeys is None or "" in apikeys:
+            key = base64.b32encode(os.urandom(60)).decode("utf-8")
```

Comment:
Any specific reason you used b32 here? Core seems to be using b64: https://github.com/opnsense/core/blob/24.1.5/src/opnsense/mvc/app/library/OPNsense/Auth/API.php#L91

Reply:
Thank you @rekup! Fixed in 2c152d5.